### PR TITLE
Add NetBSD Support.

### PR DESCRIPTION
**Pull request by fraggerfox**
Add support for NetBSD. (Addresses #301)

A lot of the work has been taken from the PR to bring up OpenBSD support. #607 

image

> NOTE: Battery and Thermal stats are shown based on some guess work.

Thanks to @0-wiz-0 for helping test and debug the NetBSD specific changes.

> NOTE: ~Once this is merged, a follow up PR will update the `README.md` with additional information for building / installing in NetBSD.~ The PR has the updated `README.md`.

## Pending changes
- [x] ~Verify the memory stats shown is correct.~
- [x]  ~Fixed the swap memory calculation.~
- [x] ~Add support for querying battery stats.~
- [x] ~Add support for querying thermal stats.~
- [x] ~Fix `utf` issues when starting up in NetBSD.~
- [x] ~Add CI for `nebtsd` based builds.~

### CODE REVIEW

**Comment by imwints** [comment]
You should also update all other CI files to exclude the new `netbsd` path.

File: src/netbsd/btop_collect.cpp [hunk 1]
```diff
@@ -0,0 +1,1437 @@
...
+		Cpu::cpuName = Cpu::get_cpuName();
+		Cpu::got_sensors = Cpu::get_sensors();
+		Cpu::core_mapping = Cpu::get_core_mapping();
+
+		//? Init for namespace Mem
+		Mem::old_uptime = system_uptime();
+		Mem::collect();
+	}
+
+	//* RAII wrapper for kvm_openfiles
```
Comment (imwints) [inline src/netbsd/btop_collect.cpp:187]: Also we could probably use the new def from `shared.hpp`

https://github.com/aristocratos/btop/blob/c6fbfd1e27c27cfe89b2e272c0419e57193ea8c5/src/btop_shared.hpp#L96-L103

Same goes for the getifaddr function.

File: src/netbsd/btop_collect.cpp [hunk 2]
```diff
@@ -0,0 +1,1437 @@
...
+	vector<string> interfaces;
+	string selected_iface;
+	int errors = 0;
+	std::unordered_map<string, uint64_t> graph_max = {{"download", {}}, {"upload", {}}};
+	std::unordered_map<string, array<int, 2>> max_count = {{"download", {}}, {"upload", {}}};
+	bool rescale = true;
+	uint64_t timestamp = 0;
+
+	//* RAII wrapper for getifaddrs
+	class getifaddr_wrapper {
```
Comment (imwints) [inline src/netbsd/btop_collect.cpp:926]: ...here

File: src/netbsd/btop_collect.cpp (hunk 1 above)
Comment (fraggerfox) [reply src/netbsd/btop_collect.cpp:187]: Let me look into this and update it.

**Comment by fraggerfox** [comment]
> You should also update all other CI files to exclude the new `netbsd` path.

Good point, let me do that update.

File: src/netbsd/btop_collect.cpp (hunk 1 above)
Comment (fraggerfox) [reply src/netbsd/btop_collect.cpp]: Should be done in the latest commit.

File: src/netbsd/btop_collect.cpp (hunk 2 above)
Comment (fraggerfox) [reply src/netbsd/btop_collect.cpp]: Should be done in the latest commit.

**Comment by fraggerfox** [comment]
> > You should also update all other CI files to exclude the new `netbsd` path.
> 
> Good point, let me do that update.

Should be done in the latest commit.

**Comment by aristocratos** [comment]
@fraggerfox

Is this PR finished?
If so, do you wanna note down in this thread if there are any specific dependencies or instructions for compiling on NetBSD.

Just to save me the trouble later when doing testing.

Any links to VM images of current NetBSD release would also be helpful.

**Comment by aristocratos** [comment]
> NOTE: Once this is merged, a follow up PR will update the README.md with additional information for building / installing in NetBSD.

I would recommend you include the README.md changes in this PR also if you don't mind, it would speed things along when I have time to look through everything and do testing.
Since I will likely be pushing a new release once this is merged.

**Comment by fraggerfox** [comment]
> @fraggerfox
> 
> Is this PR finished? If so, do you wanna note down in this thread if there are any specific dependencies or instructions for compiling on NetBSD.
> 
> Just to save me the trouble later when doing testing.
> 
> Any links to VM images of current NetBSD release would also be helpful.

I have added an updated `README.md` with the instructions for NetBSD.

You can find the VM image which I have used from here
- https://github.com/vmactions

but if you want to load it up in a Virtualbox like environment, the ISO from NetBSD should help out
- https://wiki.netbsd.org/ports/amd64/

**Comment by fraggerfox** [comment]
@aristocratos Do let me know if any additional information / changes are needed from my end to help merge this PR? It will also help us to finish off the package for this in NetBSD. Thank you!

**Review by aristocratos — APPROVED** [review]
Seems to be working great :)

Big thanks for contributing!

**Pull request merged**